### ISSUE TITLE
Deepfool attack simply returns the Gradient

### ISSUE THREAD

**Issue by tkupek** [bethgelab/foolbox#426]
For me, the Deepfool attack is not working as expected.
While I'm getting good results for FGSM and BIM, Deepfool only outputs the Gradient of the image.

The bounds + preprocessing is exactly the same as for the other two attacks. What am I missing?

```
fmodel = foolbox.models.TensorFlowEagerModel(model, bounds=(0, 1))
attack = foolbox.attacks.DeepFoolAttack(fmodel)

images, labels = foolbox.utils.samples('mnist')
images = images.reshape(1, 28, 28, 1) / 255
adversarial = attack(images, labels, steps=3)
```

Using Tensorflow 2.0.0 with foolbox 2.3.0

**Comment by jonasrauber** [comment]
Is this still relevant?

**Comment by tkupek** [comment]
Yes, just retestet this and the issue seems to persist.

**Comment by jonasrauber** [comment]
Sorry, I don't get it. Could you share a minimal (but executable) example with expected and actual outcome.

**Comment by tkupek** [comment]
Sorry if my description was not sufficient to address this issue. Please find a minimal execution example below. You will need a pretrained model to run it.

```
import os
import matplotlib.pyplot as plt
import numpy as np
import foolbox
from keras.datasets import mnist
from tensorflow.keras.models import load_model


(_, _), (x_test, y_test) = mnist.load_data()
x_test = x_test[:10]
y_test = y_test[:10]

x_test = x_test.astype('float32') / 255.0
x_test = x_test.reshape(x_test.shape[0], x_test.shape[1], x_test.shape[2], 1)
y_test = y_test.astype(np.int64)

MODEL_PATH = os.path.join('tmp', 'pretrained', 'pretrained-lenet-mnist-0.h5')
model = load_model(MODEL_PATH)

# plot original image
plt.imshow(np.squeeze(x_test[0]))
plt.show()

# FGSM attack
fmodel = foolbox.models.TensorFlowEagerModel(model, bounds=(0, 1))
attack = foolbox.attacks.GradientSignAttack(fmodel)
adv = attack(x_test, y_test, epsilons=[0.5], max_epsilon=1.0)

plt.imshow(np.squeeze(adv[0]))
plt.show()

# Deepfool attack
fmodel = foolbox.models.TensorFlowEagerModel(model, bounds=(0, 1))
attack = foolbox.attacks.DeepFoolAttack(fmodel)
adv = attack(x_test, y_test, steps=5)

plt.imshow(np.squeeze(adv[0]))
plt.show()
```

The output are three images, the original one, FGSM attack and Deepfool attack:
<img src="https://user-images.githubusercontent.com/4039384/74140465-6615d900-4bf5-11ea-9dbc-d1e05baa3a55.png" width="200">
<img src="https://user-images.githubusercontent.com/4039384/74140463-657d4280-4bf5-11ea-9d96-c7781425c970.png" width="200">
<img src="https://user-images.githubusercontent.com/4039384/74140462-64e4ac00-4bf5-11ea-8df0-8b17b38717ab.png" width="200">

While the FGSM looks good, the Deepfool attack does not work as expected.
I hope this clarifies my problem. If you need further explanations please let me know.

**Comment by zimmerrol** [comment]
Closing this issue as it relates to an earlier version of foolbox.
